File: sd/trabalhoFinal/final/despachante.py
import json
from esqueleto import Esqueleto
from filme import Filme

class Despachante:
    def __init__(self):
        self.esqueleto = Esqueleto()

    def invoke(self, request):
        request_data = json.loads(request)
        methodId = request_data.get("methodId")

        if methodId == 1:
            response = self.esqueleto.adicionar_filme(request_data)
        elif methodId == 2:
            response = self.esqueleto.remover_filme(request_data)
        elif methodId == 3:
            response = self.esqueleto.exibir_detalhe(request_data)
        elif methodId == 4:
            response = self.esqueleto.mostrar_catalogo()

        response_data = {
            "messageType": 0,
            "objectreference": request_data.get("objectreference"),
            "methodId": methodId,
            "arguments": response.get("result"),
            }
        response = json.dumps(response_data)

        return response

File: sd/trabalhoFinal/final/UDPServer/src/main/java/org/example/Despachante.java
package org.example;
import org.json.JSONException;
import org.json.JSONObject;
import java.sql.SQLException;

public class Despachante {
    private Esqueleto esqueleto;

    public Despachante(Database db) throws SQLException {
        this.esqueleto = new Esqueleto(db);
    }

    public String invoke(String objectReference, int methodId, String arguments) {
        try {
            String response;
            if (methodId == 1) {
                response = esqueleto.adicionarFilme(arguments);
            } else if (methodId == 2) {
                response = esqueleto.removerFilme(arguments);
            } else if (methodId == 3) {
                response = esqueleto.exibirDetalhe(arguments);
            } else if (methodId == 4) {
                response = esqueleto.mostrarCatalogo();
            } else {
                response = "Método não reconhecido";
            }

            return response;
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }
}
